labs/lab2/cord.cpp

```diff
@@ -49,19 +49,48 @@ cord_t cord_join(cord_t R, cord_t S) {
 
 string cord_tostring(cord_t R) {
     // TODO: Your implementation here
-    return "";
+    if (!R->len)
```

Comment:
What if `R == nullptr`? This will dereference a null pointer and access its `len` field.
```suggestion
    if (R == nullptr)
```